File: parkhaus.php

<?php
class Parkhaus {
    private $parkhausNr;
    private $ebenen;
    private $plaetzeProEbene;
    private $parkplaetze;

    // Konstruktor
    public function __construct($parkhausNr, $ebenen, $plaetzeProEbene) {
        $this->parkhausNr = $parkhausNr;
        $this->ebenen = $ebenen;
        $this->plaetzeProEbene = $plaetzeProEbene;
        $this->parkplaetze = array();

        for ($i = 0; $i < $ebenen; $i++) {
            $reihe = array();
            for ($j = 0; $j < $plaetzeProEbene; $j++) {
                $reihe[] = false;
            }
            $this->parkplaetze[] = $reihe;
        }
    }

    // Getter für Parkhausnummer
    public function getParkhausNr() {
        return $this->parkhausNr;
    }

    // Belege einen spezifischen Platz auf einer Ebene
    public function parkplatzBelegt($ebene, $platz) {
        if (!$this->parkplaetze[$ebene][$platz]) {
            $this->parkplaetze[$ebene][$platz] = true;
            return true;
        } else {
            return false;
        }
    }

    // Sperre die ganze Ebene
    public function sperreEbene($ebene) {
        for ($i = 0; $i < count($this->parkplaetze[$ebene]); $i++) {
            $this->parkplaetze[$ebene][$i] = true;
        }
    }

    // Belege einen beliebigen Platz auf einer Ebene
    public function parkplatzBelegtEbene($ebene) {
        for ($i = 0; $i < count($this->parkplaetze[$ebene]); $i++) {
            if (!$this->parkplaetze[$ebene][$i]) {
                $this->parkplaetze[$ebene][$i] = true;
                return true;
            }
        }
        return false;
    }

    // Gebe nächsten freien Platz zurück
    public function zeigeNaechstenPlatz() {
        for ($i = 0; $i < count($this->parkplaetze); $i++) {
            for ($j = 0; $j < count($this->parkplaetze[$i]); $j++) {
                if (!$this->parkplaetze[$i][$j]) {
                    $this->parkplaetze[$i][$j] = true;
                    return array($i, $j);
                }
            }
        }
        return null; // Falls kein freier Platz vorhanden ist
    }
}
?>

File: parkhaus.py
class Parkhaus:
    # Konstruktor
    def __init__(self, parkhaus_nr, ebene, plaetze_pro_ebene):
        self.__parkhaus_nr = parkhaus_nr
        self.__ebenen = ebene
        self.__plaetze_pro_ebene = plaetze_pro_ebene
        self.__parkplaetze = []
          
        for x in range(self.__ebenen):
            reihe = []
            for y in range(self.__plaetze_pro_ebene):
                reihe.append(False)
            self.__parkplaetze.append(reihe)
    
    # Getter für Parkhausnummer
    def getParkhausNr(self):
        return self.__parkhaus_nr

    # Belege einen spezifischen Patz auf einer Ebene
    def parkplatzBelegt(self,ebene,platz):
        if self.__parkplaetze[ebene][platz] == False:
            self.__parkplaetze[ebene][platz] = True
            return True
        else:
            return False
    
    # Sperre die ganze Ebene  
    def sperreEbene(self,ebene):
        for i in range(len(self.__parkplaetze[ebene])):
            self.__parkplaetze[ebene][i] = True

    #Belge einen beliebigen Platz auf einer Ebene
    def parkplatzBelegtEbene(self,ebene):
        for i in range(len(self.__parkplaetze[ebene])):
            if self.__parkplaetze[ebene][i] == False:
                self.__parkplaetze[ebene][i] = True
                return True
                break
        return False

    # Gebe nächsten freien Platz zurück
    def zeigeNaechstenPlatz(self):
        freierPlatz = []
        for i in range(len(self.__parkplaetze)):
            for j in range(len(self.__parkplaetze[i])):
                if self.__parkplaetze[i][j] == False:
                    self.__parkplaetze[i][j] = True
                    freierPlatz.append(i)
                    freierPlatz.append(j)
                    break
            break
        return freierPlatz
